### ISSUE TITLE
MIPS compiler adds unwanted code for handling divide by zero

### ISSUE THREAD

**Issue by AesirHod** [decompme/decomp.me#742]
The line:
delta = ((curWater->zValueTo - curWater->zValueFrom) * (curWater->curStep >> 12)) / (curWater->maxSteps >> 12);

Produces:
 > 88:    bnez    v0,94 ~>
 > 8c:    nop     
 > 90:    break   0x7
 > 94: ~> li      at,-1
 > 98:    bne     v0,at,ac ~>
 > 9c:    lui     at,0x8000
 > a0:    bne     v1,at,ac ~>
 > a4:    nop     
 > a8:    break   0x6
 
 Where the target assembly was simply:
  7c:    div     zero,v1,v0
  
  From the discussion on discord, this is preventable in some compilers, but the required flags aren't currently handled by decomp.me.

**Comment by mkst** [comment]
Out of interest, what are the flags?

**Comment by simonlindholm** [comment]
This seems like a matter for the assembler and not the compiler, so not surprising that cc1psx is unhappy.

**Comment by mkst** [comment]
Yeah, need to feed them into gcc and see what flags it tries to pass on to aspsx

**Comment by mkst** [comment]
I'm wondering if the source assembly only contains the pseudo op rather than the full list of commands?

**Comment by mkst** [comment]
Closing as cant/wontfix. will reopen if more information becomes available.

**Comment by mkst** [comment]
The solution is to pass `-0` to ASPSX.EXE. Currently decomp.me does not allow for custom flags to be sent to the assembly portion of a compilation - and the PSYQ compilers invoke CC1PSX.EXE directly (rather than using CCPSX.EXE wrapper) so we are a bit stuck.
